sfge/src/main/java/com/salesforce/graph/build/VertexSynthesisUtil.java

```diff
@@ -0,0 +1,108 @@
+package com.salesforce.graph.build;
+
+import com.salesforce.apex.jorje.ASTConstants.NodeType;
+import com.salesforce.collections.CollectionUtil;
+import com.salesforce.exception.ProgrammingException;
+import com.salesforce.graph.Schema;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeSet;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+
+/**
+ * Utility class for the standardized addition of properties to Synthetic graph vertices.
+ *
+ * <p>TODO: Parts of {@link StaticBlockUtil} can be refactored into this class.
+ *
+ * <p>TODO: This class has redundant code that could stand to be cleaned up.
+ */
+public class VertexSynthesisUtil {
```

Comment:
A lot of the methods in here are lightly modified copies of the ones in `StaticBlockUtil`.